doc/CMakeLists.txt

```diff
@@ -0,0 +1,89 @@
+
+cmake_minimum_required(VERSION 3.8.2)
+project(Zephyr-Kernel-Doc VERSION ${PROJECT_VERSION})
+
+set(ZEPHYR_BASE $ENV{ZEPHYR_BASE})
+
+message(STATUS "Zephyr base: ${ZEPHYR_BASE}")
+
+find_package(PythonInterp 3.4)
+set(DOXYGEN_SKIP_DOT True)
+find_package(Doxygen)
+
+find_program(
+  SPHINXBUILD
+  sphinx-build
+  )
+if(${SPHINXBUILD} STREQUAL SPHINXBUILD-NOTFOUND)
+  message(FATAL_ERROR "The 'sphinx-build' command was not found. Make sure you have Sphinx installed.")
+endif()
+
+if(NOT DEFINED SPHINXOPTS)
+  set(SPHINXOPTS -q)
+endif()
+
+if(NOT DEFINED DOC_TAG)
+  set(DOC_TAG development)
+endif()
+
+# Internal variables.
+set(ALLSPHINXOPTS  -d ${CMAKE_BINARY_DIR}/doctrees ${SPHINXOPTS} .)
+# the i18n builder cannot share the environment and doctrees with the others
+set(I18NSPHINXOPTS  ${SPHINXOPTS} .)
+
+set(DOXYFILE ${CMAKE_BINARY_DIR}/build.doxyfile)
+
+set_property(SOURCE APPEND PROPERTY CMAKE_CONFIGURE_DEPENDS ${DOXYFILE})
```

Comment:
This comment covers the lines: 36-38:
You can use configure_file instead.
Advantage of configure_file is that it will setup the CMake dependency for you as well.

> `set_property(SOURCE APPEND PROPERTY CMAKE_CONFIGURE_DEPENDS ${DOXYFILE})`
> `file(READ ${CMAKE_CURRENT_SOURCE_DIR}/zephyr.doxyfile CONTENTS)`
> `file(WRITE ${DOXYFILE} ${CONTENTS})`

can be written as:
`configure_file(${CMAKE_CURRENT_SOURCE_DIR}/zephyr.doxyfile ${DOXYFILE} COPYONLY)`

and then you just append to the file, as you do now with:
> `file(APPEND ${DOXYFILE} "STRIP_FROM_PATH=${ZEPHYR_BASE}")`